Core/Core/Configuration/Config/SegmentConfig.swift

```diff
@@ -0,0 +1,31 @@
+//
+//  SegmentConfig.swift
+//  Core
+//
+//  Created by Anton Yarmolenka on 02/02/2024.
+//
+
+import Foundation
+
+private enum SegmentKeys: String, RawStringExtractable {
+    case enabled = "ENABLED"
+    case writeKey = "SEGMENT_IO_WRITE_KEY"
+}
+
+public final class SegmentConfig: NSObject {
+    public var enabled: Bool = false
+    public var writeKey: String = ""
+    
+    init(dictionary: [String: AnyObject]) {
+        super.init()
+        enabled = dictionary[SegmentKeys.enabled] as? Bool == true
+        writeKey = dictionary[SegmentKeys.writeKey] as? String ?? ""
```

Comment:
Let's make `enabled` also dependent on `writeKey` like the writeKey shouldn't be empty.

Reply:
done